### PR TITLE
Implement event sending

### PR DESCRIPTION
**Pull request by kittylyst**

### CODE REVIEW

File: telemetry/src/test/java/com/newrelic/telemetry/TelemetryClientTest.java [hunk 1]
```diff
@@ -49,10 +49,11 @@ private MetricBatch makeBatch(Collection<Metric> metrics) {
   @Test
   void sendHappyPath() throws Exception {
```
Comment (jkwatson) [inline telemetry/src/test/java/com/newrelic/telemetry/TelemetryClientTest.java:50]: we should add a "happy path" test for events as well.

File: telemetry_core/src/main/java/com/newrelic/telemetry/AbstractSenderBuilder.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
+package com.newrelic.telemetry;
+
+import com.newrelic.telemetry.http.HttpPoster;
+import com.newrelic.telemetry.util.Utils;
+import java.io.UncheckedIOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+
+public abstract class AbstractSenderBuilder<B extends AbstractSenderBuilder<B>> {
```
Comment (jkwatson) [inline telemetry_core/src/main/java/com/newrelic/telemetry/AbstractSenderBuilder.java:10]: I get a twitchy when I see abstract parameterized builders. Especially when the `build()` method ends up returning `Object`. Let's create an issue, to be done quickly after this is merged, get rid of the templated builder and replace it with something better. I would bet that @breedx-nr has all the solutions to making it great in his back pocket.

File: telemetry_core/src/main/java/com/newrelic/telemetry/events/EventBuffer.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+   * EventBatch}.
+   *
+   * @return A new {@link EventBatch} with an immutable collection of {@link Event Events}.
+   */
+  public EventBatch createBatch() {
+    logger.debug("Creating Event batch.");
+    Collection<Event> eventsForBatch = new ArrayList<>(this.events.size());
+
+    // Drain the Event buffer and return the batch
+    Event ev;
```
Comment (jkwatson) [inline telemetry_core/src/main/java/com/newrelic/telemetry/events/EventBuffer.java:64]: rename to "event"

File: telemetry_core/src/main/java/com/newrelic/telemetry/transport/BatchDataSender.java [hunk 4]
```diff
@@ -116,7 +116,7 @@ private Response sendPayload(byte[] payload)
           "Response from New Relic ingest API: code: {}, body: {}",
           response.getCode(),
           response.getBody());
-      if (response.getCode() == 202) {
+      if (response.getCode() == 202 || response.getCode() == 200) {
```
Comment (jkwatson) [inline telemetry_core/src/main/java/com/newrelic/telemetry/transport/BatchDataSender.java:119]: might be worth a comment here about the 200 being a special case for the current event API

File: telemetry_examples/src/main/java/com/newrelic/telemetry/examples/TelemetryClientExample.java [hunk 5]
```diff
@@ -28,25 +32,40 @@
...
     String insightsInsertKey = args[0];
 
-    MetricBatchSender batchSender =
+    MetricBatchSender metricBatchSender =
         SimpleMetricBatchSender.builder(insightsInsertKey, Duration.of(10, ChronoUnit.SECONDS))
             .build();
     SpanBatchSender spanBatchSender = SimpleSpanBatchSender.builder(insightsInsertKey).build();
 
-    TelemetryClient telemetryClient = new TelemetryClient(batchSender, spanBatchSender);
+    // todo: make a simple version of this.
```
Comment (jkwatson) [inline telemetry_examples/src/main/java/com/newrelic/telemetry/examples/TelemetryClientExample.java:40]: let's create an issue for this, as well, so we don't lose track.

Comment (kittylyst) [reply telemetry_examples/src/main/java/com/newrelic/telemetry/examples/TelemetryClientExample.java:40]: #140 filed to track this.

File: telemetry_core/src/main/java/com/newrelic/telemetry/AbstractSenderBuilder.java (hunk 2 above)
Comment (kittylyst) [reply telemetry_core/src/main/java/com/newrelic/telemetry/AbstractSenderBuilder.java:10]: #139 filed to track this.

File: telemetry/src/main/java/com/newrelic/telemetry/TelemetryClient.java [hunk 6]
```diff
@@ -68,6 +76,15 @@ public void sendBatch(SpanBatch batch) {
     scheduleBatchSend((b) -> spanBatchSender.sendBatch((SpanBatch) b), batch, 0, TimeUnit.SECONDS);
   }
 
+  /**
+   * Send a batch of metrics, with standard retry logic. This happens on a background thread,
```
Comment (jkwatson) [inline telemetry/src/main/java/com/newrelic/telemetry/TelemetryClient.java:80]:
```suggestion
   * Send a batch of events, with standard retry logic. This happens on a background thread,
```

Comment (kittylyst) [reply telemetry/src/main/java/com/newrelic/telemetry/TelemetryClient.java:80]: This is resolved in the other PR :)

Comment (jkwatson) [reply telemetry/src/main/java/com/newrelic/telemetry/TelemetryClient.java:80]: aha. I got confused with the PR inception.

**Comment by breedx-nr** [comment]
Thanks again!

**Pull request merged**